californium-core/src/test/java/org/eclipse/californium/core/test/BlockwiseTransferTest.java

```diff
@@ -220,42 +222,56 @@ public void onCancel() {
 	 * <p>The targeted endpoint has the {@link NetworkConfig.Keys#BLOCKWISE_STRICT_BLOCK2_OPTION} set to true and should respond with a block2 option indicating that no more blocks are available. </p>
 	 * 
 	 * @throws InterruptedException
 	 */
 	@Test
-	public void testEarlyNegociationWithStrictBlock2() throws InterruptedException {
+	public void testEarlyNegotiationWithStrictBlock2() throws InterruptedException {
 		
-		testGetRequestWithEarlyNegocition(true, PARAM_SHORT_RESP);
+		testGetRequestWithEarlyNegotiation(true, PARAM_SHORT_RESP);
 	}
 	
 	/**
-	 * Send request to the server with early blockwise negociation through block2 option. The response content should fits into a single response.
+	 * Send request to the server with early blockwise negotiation through block2 option. The response content should fits into a single response.
 	 * <p>The targeted endpoint has the {@link NetworkConfig.Keys#BLOCKWISE_STRICT_BLOCK2_OPTION} set to false and should respond without a block2 option. </p>
 	 * 
 	 * @throws InterruptedException
 	 */
 	@Test
-	public void testEarlyNegociationWithoutStrictBlock2() throws InterruptedException {
+	public void testEarlyNegotiationWithoutStrictBlock2() throws InterruptedException {
 	
-		testGetRequestWithEarlyNegocition(false, PARAM_SHORT_RESP);
+		testGetRequestWithEarlyNegotiation(false, PARAM_SHORT_RESP);
 	}
 	
 	/**
-	 * Send request to the server with early blockwise negociation through block2 option. The response content should be empty and contains the block2 option.
+	 * Send request to the server with early blockwise negotiation through block2 option. The response content should be empty and contains the block2 option.
 	 * <p>The targeted endpoint has the {@link NetworkConfig.Keys#BLOCKWISE_STRICT_BLOCK2_OPTION} set to true and should respond with a block2 option. </p>
 	 * 
 	 * @throws InterruptedException
 	 */
 	@Test
-	public void testEarlyNegociationWithStrictBlock2NoResponsePayload() throws InterruptedException {
+	public void testEarlyNegotiationWithStrictBlock2NoResponsePayload() throws InterruptedException {
+	
+		testGetRequestWithEarlyNegotiation(true, PARAM_EMPTY_RESP);
+	}
 	
-		testGetRequestWithEarlyNegocition(true, PARAM_EMPTY_RESP);
+	
+	/**
+	 * Test that consecutive requests to a same resource with early blockwise negotiation are both going through the applicative layer
+	 * @throws InterruptedException
+	 */
+	@Test
+	public void testMultipleEarlyNegotiationWithShortInterval() throws InterruptedException {
+		final int nbApplicativeRequestReceived = applicativeLayerGetRequestCount.get();
+		testGetRequestWithEarlyNegotiation(false, PARAM_SHORT_RESP);
+		testGetRequestWithEarlyNegotiation(false, PARAM_SHORT_RESP);
+		
+		assertEquals("Applicative layer received only one request", 2, applicativeLayerGetRequestCount.get() - nbApplicativeRequestReceived);
```

Comment:
I would prefer a implementation, which calls `applicativeLayerGetRequestCount.set(0);` in the `@After`instead of the calculation.

Reply:
May be the text "Application layer received not two requests" would describe the failure not the expectation (what, if 3 is returned :-) ).

Reply:
I'm not sure, why you use "applicative", I would use "application".

Reply:
> I would prefer a implementation, which calls `applicativeLayerGetRequestCount.set(0);` in the `@After`instead of the calculation.

Sorry but I'm not sure to understand what it would looks like, from what I know the `@After`method is called last to clean the test context. I committed a new version which you may prefer.

> May be the text "Application layer received not two requests" would describe the failure not the expectation (what, if 3 is returned :-) ).

Yes you're right ! I edited the message.

> I'm not sure, why you use "applicative", I would use "application".

A bad translation from French to English :-).

Reply:
> Sorry but I'm not sure to understand what it would looks like, from what I know the `@After`method is called last to clean the test context.

Yes, the `BlockwiseTransferTest` contains several tests, and `@After` is executed after each to cleanup. If it's there, and someone adds a tests, which uses your counter again, that test get's also a "clean" counter. 

With the updated approach, every tests, which uses the counter, must reset it by it's own.

```
@After
public void destroyClient() throws Exception {
	clientEndpoint.destroy();
	applicativeLayerGetRequestCount.set(0);
}
```

Reply:
Thanks, I should have made more test on my own and I would have seen that its called after every `@Test`. I had a wrong understanding and thought it was only called only once after all tests were run.

I fixed it and added destroy on the second client endpoint too. Should be ok this time :-)

Reply:
@BeforeClass - once per class
@Before - once per test
@Test - the tests
@After - once per test
@AfterClass - once per class